### PR TITLE
Created an SSE route for the API

### PR DESCRIPTION
**Pull request by seanrichardson95**
Created a route, "/streams" that will initiate a new SSE connection and start streaming flag data.

Added the `updateStream` middleware to the update, delete, and add flag routes so that updates to the flags state will be received by those connected to our API via the SSE.

Altered `createFlag`, `editFlag` and `deleteFlag` methods on the `flagsController` to set state onto the request object and to call `next()` after sending back a response.

Created a `streamController` to house all the functions that deal with the stream and to hold the state of the stream.

### CODE REVIEW

**Comment by calebrs** [comment]
Very nice use of middleware!

So, we send the client all the flag data when it changes? I know it's probably a later stage implementation, but would it be possible to send the client only the flags that apply to them? It might help to reduce the amount of data the client receives. Again, I think that's a part of implementing the "app" database entity.

Great job on this. I'm cool with a merge.

File: server/controllers/streamController.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  clients.push(newClient);
+  req.on('close', () => {
+    console.log(`${clientId} Connection closed`);
+    clients = clients.filter(client => client.id !== clientId);
+  });
+}
+
+const sendUpdate = (req, res, next) => {
+  const updatedFlags = updateFlags(req, [...flags]);
+  if (updateFlags) {
```
Comment (julia-martin) [inline server/controllers/streamController.js]: what is the `if (updateFlags)` doing? Isn't that a function? Did you maybe mean `updatedFlags`?

File: server/routes/api.js [hunk 2]
```diff
@@ -9,10 +10,14 @@ router.get("/flags", flagsController.getAllFlags);
 // router.get("/flags", validateSDKKey, flagsController.getAllFlags);
 router.get("/flags/:id", flagsController.getFlag);
 
-router.post("/flags", validateNewFlag, flagsController.createFlag);
+router.get("/stream", streamController.handleNewConnection)
 
-router.put("/flags/:id", flagsController.editFlag);
+router.get("/status", streamController.status);
```
Comment (julia-martin) [inline server/routes/api.js:15]: Can you remind me what the GET /status endpoint is for?

**Comment by julia-martin** [comment]
Great job and I like how organized everything is! I just had a few questions on things to clarify, I'm probably just missing something simple

File: server/routes/api.js (hunk 2 above)
Comment (seanrichardson95) [reply server/routes/api.js:15]: Get status is just to let us know how many people are currently connected in the SSE connection.

File: server/controllers/streamController.js (hunk 1 above)
Comment (seanrichardson95) [reply server/controllers/streamController.js]: Whoops, good catch!

**Comment by seanrichardson95** [comment]
@calebrs Yeah, I can change the functionality to that later once we know how specifically we're implementing it on the SDK. For now I just wanted to make sure we had all the proper flag data sent through.

**Pull request merged**